packages/api/src/services/microsoft.calendar.ts

```diff
@@ -43,12 +43,23 @@ export class MicrosoftCalendarService {
     this.graphClient = Client.initWithMiddleware({
       authProvider: {
         getAccessToken: async () => {
-          const result = await this.msalClient.acquireTokenByRefreshToken({
-            refreshToken: this.refreshToken,
-            scopes: azureScopes,
-          });
-          // TODO: Save the refresh token to our DB if it is changed (result.refreshToken)
-          return result!.accessToken;
+          try {
+            const result = await this.msalClient.acquireTokenByRefreshToken({
+              refreshToken: this.refreshToken,
+              scopes: azureScopes,
+            });
+            console.log("access token refreshed");
+            console.log(result);
+            // TODO: Save the refresh token to our DB if it is changed (result.refreshToken)
+            return result!.accessToken;
+          } catch (cause) {
+            console.error("Error while refreshing access token", cause);
+            throw new TRPCError({
+              code: "INTERNAL_SERVER_ERROR",
+              message: "Error while refreshing access token",
+              cause,
+            });
+          }
```

Comment:
the microsoft service needs testing with a business account, which i dont have

Reply:
You have `julius@agreetoapp.onmicrosoft.com` -- does this one not work?

It's on the business subscription.